tests/std/tests/P0811R3_midpoint_lerp/test.cpp

```diff
@@ -1023,6 +1023,15 @@ bool test_lerp() {
     return true;
 }
 
+void test_gh_2112() {

```

Comment:
I would additionally like to see a small amount of test coverage for the type computation - basically ensuring that `decltype(lerp(args))` is `double` when a mix of types without `long double` is involved, but `long double` when any `long double` appears. It doesn't need to be super exhaustive, but we should test a `long double` in every position, e.g.

`int, int, int => double`
`long double, int, int => long double`
`int, long double, int => long double`
`int, int, long double => long double`